nixos/tests/cockroachdb.nix

```diff
@@ -0,0 +1,24 @@
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "cockroachdb";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ ];
+  };
+
+  nodes = {
+    master =
+      { pkgs, config, ... }:
+
+      {
+        services.cockroachdb.enable = true;
+        services.cockroachdb.package = pkgs.cockroachdb;
+      };
+  };
+
+  testScript = ''
+    startAll;
+
+    $master->waitForUnit("cockroachdb");
+    $master->sleep(10); # Hopefully this is long enough!!
```

Comment:
If you put `Type=notify` in the systemd service file then systemd should wait until the port is bound and you wouldn't need to sleep.